app/controllers/facilities_controller.rb

```diff
@@ -1,46 +1,51 @@
 class FacilitiesController < ApplicationController
-    def index
-        @facilities = Facility.all
-    end
+  before_action :authenticate_user!, only: [:edit, :update, :destroy, :base_params, :create_params, :update_params]
+  def index
+    @facilities = Facility.all
+  end
     
-    def new
-        @facility = Facility.new
-    end
+  def new
+    @facility = Facility.new
+  end
     
-    def create
-        facility = Facility.new
-        facility.name = params[:facility][:name]
-        facility.city = params[:facility][:city]
-        facility.state = params[:facility][:state]
-        facility.zip = params[:facility][:zip]
-        facility.specialty = params[:facility][:specialty]
-        facility.save
-        redirect_to "/"
-        
-    end
+  def create
+    @facility = Facility.create(create_params)
+    @facility.save
+  redirect_to "/"
+      
+  end
     
-    def edit
-        @facility = Facility.find(params[:id])
-    end
+  def edit
+    @facility = Facility.find(params[:id])
+  end
     
-    def update
-        facility = Facility.find(params[:id])
-        facility.name = params[:facility][:name]
-        facility.city = params[:facility][:city]
-        facility.state = params[:facility][:state]
-        facility.zip = params[:facility][:zip]
-        facility.specialty = params[:facility][:specialty]
-        facility.save
-        redirect_to "/"
-    end
+  def update
+    @facility = Facility.find(params[:id])
+    @facility.update(update_params)
+    @facility.save
+    redirect_to "/admins"
```

Comment:
[Style]
The slightly more Rails-y way of doing this is to call the path (e.g., `admins_path`).

Reply:
Changed "/admins" to "admins_url" in admins, items, procedure and facilities controllers.